src/array/struct_.rs

```diff
@@ -134,10 +135,37 @@ impl StructArray {
         &self.values
     }
 
+    /// Returns the field at `pos`.
+    pub fn value(&self, pos: usize) -> &ArrayRef {
+        &self.values[pos]
+    }
+
+    /// Return the number of fields in this struct array
+    pub fn num_columns(&self) -> usize {
+        self.values.len()
+    }
+
     /// Returns the fields of this [`StructArray`].
     pub fn fields(&self) -> &[Field] {
         Self::get_fields(&self.data_type)
     }
+
+    /// Return field names in this struct array
+    pub fn column_names(&self) -> Vec<&str> {
```

Comment:
This operation is `O(F)` where `F` is the number of fields and requires an allocation. Generally, it is more efficient to use `self.fields().iter()`.